### PR TITLE
Simplify implementation of newWithStorage1d.

### PR DESCRIPTION
**Pull request by gchanan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #34403 Port set_ to ATen.
* #34389 Stop creating unnecessary Storage with newWithStorage1d.
* #34388 Kill newWithStorage.
* #34387 Get rid of newWithSize.
* #34386 Stop last usage of newWithSize.
* #34385 Kill some unused TH(C)Storage functions.
* #34384 Stop calling newWithSize directly.
* #34383 Clean up unused newWithSize variants.
* **#34382 Simplify implementation of newWithStorage1d.**

The previous implementation was handling both newWithStorage and newWithSize, which doesn't make much sense.

Differential Revision: [D20311056](https://our.internmc.facebook.com/intern/diff/D20311056)

### CODE REVIEW

**Comment by gchanan** [comment]
This is just to separate out the handling of "has a storage" and "doesn't have a storage" cases.

**Review by bhosmer — COMMENTED** [review]
Looks like you've just so to speak inlined the 1d case, right? Is it for perf?

File: aten/src/TH/generic/THTensor.cpp [hunk 1]
```diff
@@ -88,7 +88,15 @@ THTensor *THTensor_(newWithStorage)(THStorage *storage, ptrdiff_t storageOffset,
 THTensor *THTensor_(newWithStorage1d)(THStorage *storage, ptrdiff_t storageOffset,
                                int64_t size0, int64_t stride0)
 {
-  return THTensor_(newWithStorage)(storage, storageOffset, {size0}, {stride0});
+  THStorage *new_storage = THStorage_(new)();
+  THTensor *self = c10::make_intrusive<at::TensorImpl, at::UndefinedTensorImpl>(
+    c10::intrusive_ptr<at::StorageImpl>::reclaim(new_storage),
+    at::DispatchKey::CPUTensorId
+  ).release();
+  THTensor_(setStorageNd)(self, storage, storageOffset, 1,
```
Comment (bhosmer) [inline aten/src/TH/generic/THTensor.cpp]: I'm noticing the unconditional `storage` here vs. the ternary in [newWithStorage](https://github.com/pytorch/pytorch/pull/34382/files#diff-e3bd8f02c06ae7de07fb1ace5bbd34eeL82), with AFAICT identical preceding logic... 

If that's actually an error, it seems like a consequence of the inlining, aka an argument for the way it was before. But maybe there are other motivations for inlining further up the stack?

**Review by bhosmer — APPROVED** [review]
Accepting bc I feel like I'm overthinking it, but see inline for a possible issue

File: aten/src/TH/generic/THTensor.cpp (hunk 1 above)
Comment (bhosmer) [reply aten/src/TH/generic/THTensor.cpp]: A: yup, answered by what's happening later in the stack

Comment (gchanan) [reply aten/src/TH/generic/THTensor.cpp]: ya sorry I didn't make that clear -- it's all handled later in the stack.  Some of this is pretty low level and can be subtle, so I wanted separate CI signal on every change.

**Pull request closed without merging**